test/run.d

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env rdmd
+/**
+Allows running tests individually
+
+Usage:
+  ./run.d <test-file>...
+
+Example:
+  ./run.d runnable/template2962.d fail_compilation/fail282.d
+
+See the README.md for all available test targets
+*/
+
+import std.algorithm, std.conv, std.datetime, std.exception, std.file, std.format,
+       std.getopt, std.parallelism, std.path, std.process, std.range, std.stdio, std.string;
+import core.stdc.stdlib : exit;
+
+const scriptDir = __FILE_FULL_PATH__.dirName;
+string resultsDir = scriptDir.buildPath("test_results");
+immutable testDirs = ["runnable", "compilable", "fail_compilation"];
+shared bool verbose; // output verbose logging
+shared bool force; // always run all tests (ignores timestamp checking)
+shared string hostDMD; // path to host DMD binary (used for building the tools)
+
+void main(string[] args)
+{
+    int jobs = totalCPUs;
+    auto res = getopt(args,
+        "j|jobs", "Specifies the number of jobs (commands) to run simultaneously (default: %d)".format(totalCPUs), &jobs,
+        "v", "Verbose command output", (cast(bool*) &verbose),
+        "f", "Force run (ignore timestamps and always run all tests)", (cast(bool*) &force),
+    );
+    if (res.helpWanted)
+    {
+        defaultGetoptPrinter(`./run.d <test-file>...
+
+Examples:
+
+    ./run.d runnable/template2962.d                              # runs a specific tests
+    ./run.d runnable/template2962.d fail_compilation/fail282.d   # runs multiple specific tests
+    ./run.d fail_compilation                                     # runs all tests in fail_compilation
+    ./run.d all                                                  # runs all tests
+    ./run.d clean                                                # remove all test results
+
+Options:
+`, res.options);
+        "\nSee the README.md for a more in-depth explanation of the test-runner.".writeln;
+        return;
+    }
+
+    // parse arguments
+    args.popFront;
+    args2Environment(args);
+
+    // allow overwrites from the environment
+    resultsDir = environment.get("RESULTS_DIR", resultsDir);
+    hostDMD = environment.get("HOST_DMD", "dmd");
+
+    // bootstrap all needed environment variables
+    auto env = getEnvironment;
+
+    // default target
+    if (!args.length)
+        args = ["all"];
+
+    alias normalizeTestName = f => format!"%s/%s"(f.absolutePath.dirName.baseName, f.baseName);
```

Comment:
`std.path.pathSeparator` instead of `/`?

Reply:
I went with buildPath